firstuseauthenticator/firstuseauthenticator.py

```diff
@@ -147,11 +147,20 @@ def authenticate(self, handler, data):
         password = data['password']
         # Don't enforce password length requirement on existing users, since that can
         # lock users out of their hubs.
-        if not self._validate_password(password) and not self._user_exists(username):
-            handler.custom_login_error = (
-                'Password too short! Please choose a password at least %d characters long.'
-                % self.min_password_length
-            )
+        with dbm.open(self.dbm_path, 'c', 0o600) as db:
+            stored_pw = db.get(username.encode(), None)
+            #check if there is no password for the user
+            # check if the password is adhering to the constraint
+            if stored_pw is None and not self._validate_password(password):
+                handler.custom_login_error = (
+                    'Password too short! Please choose a password at least %d characters long.'
+                    % self.min_password_length
+                )    
+            elif not self._validate_password(password) and not self._user_exists(username):
```

Comment:
This `else` branch is covered in the `if` one because when the user isn't in the database, `stored_pw` will be None, so the `if` one should be enough.

Also, the snippet bellow is the part that actually throws the error and and instructs the user to try again with a longer password, so it should be moved inside the conditional.
```python
self.log.error(handler.custom_login_error)
return None
```

Reply:
what about the new user who is regustering, is it being handled by some other function?

Reply:
The user and password are added to the database [here](https://github.com/jupyterhub/firstuseauthenticator/blob/master/firstuseauthenticator/firstuseauthenticator.py#L164).